packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -256,30 +256,13 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
    * @returns {Db} The database.
    */
   private db(name: string, dbOptions: DbOptions = {}): Db {
-    // Usually, we'd let the driver do the caching of the Database objects, but
-    // due to our particular needs, we need to cache DBs both by name *and* by
-    // options (so that subsequent calls to this method have their db options
-    // respected, instead of receiving the instance that was cached only by name
-    // and might have been configured with different options, e.g. read/write
-    // concern, read preference, etc.).
-    // We still need caching, though, since otherwise every call to
-    // this method creates a new driver Database object that registers itself
-    // and effectively cannot be garbage collected on its own, which would
-    // lead to memory leaks. We thus maintain our own cache.
     const key = `${name}-${JSON.stringify(dbOptions)}`;
     const dbcache = this.getDBCache();
     const cached = dbcache.get(key);
     if (cached) {
       return cached;
     }
-    const optionsWithForceNewInstace: DbOptions = {
-      ...dbOptions,
-
-      // Ensure that we actually get a fresh database instance from the driver.
-      returnNonCachedInstance: true
-    } as any; // TODO: returnNonCachedInstance supported? see NODE-2931
```

Comment:
I think for now what we could probably do is to remove the `as any` along with `: DbOptions`? That should let typescript figure out that this type is “`DbOptions` + this one other thing that is not mentioned there but that the `.db()` method still accepts”

Reply:
I think we can just put this PR on hold until it's merged into node master, I didn't realize it hadn't been already done. Then we can revert some of the tests so that it makes sure the behavior's still happening

Reply:
Until this is merged: https://github.com/mongodb/node-mongodb-native/pull/2672